kubectl-cloudflow/version/version.go

```diff
@@ -30,21 +30,47 @@ const ProtocolVersionKey = "protocol-version"
 // ProtocolVersionConfigMapName is the name of the configmap that contains the protocol-version
 const ProtocolVersionConfigMapName = "cloudflow-protocol-version"
 
-// FailOnProtocolVersionMismatch fails and exits if the protocol version of kubectl-cloudflow does not match with the cloudflow operator protocol version.
-func FailOnProtocolVersionMismatch() {
+// GetProtocolVersionConfigMap Get the protocol version config map set by the operator
+func GetProtocolVersionConfigMap() (*corev1.ConfigMap, error) {
 	k8sClient, k8sErr := k8s.GetClient()
 	if k8sErr != nil {
 		util.LogAndExit("Failed to create new kubernetes client, %s", k8sErr.Error())
 	}
 
 	var cm *corev1.ConfigMap
-	if configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{}); err == nil {
+	configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{})
+	if err == nil {
 		for i, v := range configMaps.Items {
 			if v.Name == ProtocolVersionConfigMapName {
 				cm = &configMaps.Items[i]
 			}
 		}
-	} else {
+	} 
+	return cm, err
+}
+
+// GetDefaultNamespace Get the default namespace set in the protocol version config map
+func GetDefaultNamespace() (string, error) {
```

Comment:
Maybe `GetCloudflowNamespace`? In the future, this could return a list to select from until you specified a default.

Reply:
will change ..

Reply:
An easier approach could be to let the user pass the namespace anyway at the cli even if we currently support one Cloudflow installation.

Reply:
The problem with passing the namespace is that that affects a number of programs where u need to handle an additional argument to the cli - this approach has less impact

Reply:
I'd prefer the label-scanning version proposed by @olofwalker . There is already another CLI change proposal in our backlog that will use the native `kubectl` namespace handling, including the `-n/--namespace` flag for selecting the namespace to deploy an app to. I don't think another namespace flag would make a lot of sense at this stage.

I do agree that in the far distant future, when K8s multi-tenancy support is further along, we might be able to support multiple cloudflow instances and only then does it get an issue of possibly finding two instances. It's a bit too early to worry about that right now though IMHO.

Reply:
@agemooij @debasishg Should the CLI at this point even work if it finds more than one installation of Cloudflow then? There will be two operators there fighting for the CR being created. I would vote for an error being printed in the CLI in that case.

Reply:
@olofwalker  Agreed